test/unit/definitions/divorce/UserEventAuthorisation.test.js

```diff
@@ -0,0 +1,73 @@
+const expect = require('chai').expect;
+const { differenceWith, intersectionWith, concat } = require('lodash');
+
+const caseEventsAuthTab = Object.assign(require('definitions/divorce/json/AuthorisationCaseEvent'), []);
+const caseEventFields = Object.assign(require('definitions/divorce/json/CaseEventToFields'), []);
+const caseFieldAuthTab = Object.assign(require('definitions/divorce/json/AuthorisationCaseField'), []);
+const caseRolesTab = Object.assign(require('definitions/divorce/json/CaseRoles'), []);
+let caseEventsActive = [];
+
+describe('Events authorisation validation', () => {
+  function matchEventFieldToAuthField(userRole, caseType) {
+    return (authFieldEntry, eventCaseField) => {
+      return eventCaseField.CaseFieldID === authFieldEntry.CaseFieldID
+        && authFieldEntry.UserRole === userRole
+        && authFieldEntry.CaseTypeID === caseType
+        && eventCaseField.CaseTypeID === caseType;
+    };
+  }
+
+  before(() => {
+    caseEventsActive = caseEventsAuthTab.filter(entry => {
+      return entry.CRUD === 'CRU' || entry.CRUD === 'RU';
+    });
+
+    // we need to exclude the Case Roles events as its not used for Field Authorisation (is User Role only)
+    caseEventsActive = differenceWith(caseEventsActive, caseRolesTab, (eventActive, caseRole) => {
+      return eventActive.UserRole === caseRole.ID;
+    });
+  });
+
+  it('should have at least CRU or RU access level for all MANDATORY, OPTIONAL and READONLY show/hide event fields', () => {
+    caseEventsActive.forEach(eventAuth => {
+      const userRole = eventAuth.UserRole;
+      const eventName = eventAuth.CaseEventID;
+      const caseType = eventAuth.CaseTypeID;
+
+      // get all MANDATORY and OPTIONAL fields for this event
+      let caseFieldsForEvent = caseEventFields.filter(entry => {
```

Comment:
can you please extract logic to named function and make sure if this name tells what function doeas

Reply:
Done 👍